src/Lecture2.hs

```diff
@@ -70,7 +79,37 @@ return the removed element.
 >>> removeAt 10 [1 .. 5]
 (Nothing,[1,2,3,4,5])
 -}
-removeAt = error "TODO"
+
+removeAt :: Int -> [a] -> (Maybe a, [a])
+removeAt i list
+   | i < 0               = (Nothing, list)
+   | null list           = (Nothing, list)
+   | i == 0              = (Just (head list), tail list)
+   | otherwise =
+      let x = (removeAt (i-1) (tail list))
+         in case x of
+            (Nothing, list1)    -> (Nothing, list)
+            (Just value, list1) -> (Just value, head list : list1)
```

Comment:
You can simplify the implementation a bit if you notice that you don't need to pattern match on `Maybe` here 🙂 
```suggestion
      let (val, list1) = removeAt (i-1) (tail list)
      in (val, head list : list1)
```